pkg/armhelpers/azureclient.go

```diff
@@ -462,3 +463,45 @@ func (az *AzureClient) addAcceptLanguages() autorest.PrepareDecorator {
 		})
 	}
 }
+
+func (az *AzureClient) addAuxiliaryTokens() autorest.PrepareDecorator {
```

Comment:
would it be possible to rename this function to something more specific so we don't have two functions with the same name?

Reply:
Happy to update it. Any recommendations? I was following the `AddAcceptLanguages` and `addAcceptLanguages` example in the file.

Reply:
how about `setAuxiliaryTokens`?

Also, is there a good reason why we're using encapsulation like this? (`AddAuxiliaryTokens` is a public method and the one we're discussing is a private method).

@tariq1890 do you have any thoughts?

Reply:
There is definitely redundancy. In the case of passing acceptLanguages and auxiliary tokens, there is actually an overwrrite of the requestPreparer happening.

Reply:
@jackfrancis I will update it `setAuxiliaryTokens`

>Also, is there a good reason why we're using encapsulation like this?

I was following the coding convention in the file that's already being used by `addAcceptLanguages`. I am not aware of the initial reason to follow this encapsulation. But it sure introduces a lot of redudancy and superflous for a seemingly simple task. I am happy to update this to call `setAuxiliaryTokens` only once.

Reply:
Updated function name + removed redundant setting of header. PTAL.